Assets/Scripts/Weapons/WeaponWithLimitedAmmo.cs

```diff
@@ -0,0 +1,27 @@
+using UnityEngine;
+
+public class WeaponWithLimitedAmmo: BaseWeapon
+{
+    public int maxAmmo;
+	private int currentAmmo;
+
+	protected new void Start()
+    {
+		currentAmmo = maxAmmo;
+
+        base.Start();
+    }
+
+
+    public override bool TryToUseWeapon()
+    {
+        if (currentAmmo > 0 && base.TryToUseWeapon())
+        {
+            --currentAmmo;
```

Comment:
Tu jeszcze trzeba by dodać jakoś przypisanie zmiennej weapon z powrotem do DefaultWeapon. Teraz jak się skończy amunicja to nie można strzelać w ogóle. Ale skoro i tak ten PR to hack to może bym to zostawił.

Reply:
Done